single-line-diagram-core/src/main/java/com/powsybl/sld/layout/HorizontalSubstationLayout.java

```diff
@@ -27,7 +27,9 @@ public HorizontalSubstationLayout(SubstationGraph graph, VoltageLevelLayoutFacto
     @Override
     protected List<Point> calculatePolylineSnakeLine(LayoutParameters layoutParam, Node node1, Node node2,
                                                      boolean increment) {
-        return calculatePolylineSnakeLineForHorizontalLayout(layoutParam, node1, node2, increment, infosNbSnakeLines);
+        double yMin = getGraph().getVoltageLevels().stream().mapToDouble(VoltageLevelGraph::getY).min().orElse(0.0);
+        double yMax = getGraph().getVoltageLevels().stream().mapToDouble(g -> g.getY() + g.getInnerHeight(layoutParam)).max().orElse(0.0);
```

Comment:
Just one thing I forgot, to avoid calculating it for each node we could calculate yMin & yMax in AbtractLayout::manageSnakeLines and pass it as arguments.

Reply:
> Just one thing I forgot, to avoid calculating it for each node we could calculate yMin & yMax in AbtractLayout::manageSnakeLines and pass it as arguments.

For VerticalLayout case, we need to compute it for each node, right ?

Reply:
yes indeed! let's forget about this, and I'll create an issue to create yMin / yMax attribute for substation graphs, or more generally for any kind of graphs